types.go

```diff
@@ -121,6 +121,15 @@ type functionInfo struct {
 	statuser         Statuser
 }
 
+func (f functionInfo) canEvaluate(grad, hess bool) error {
+	if grad && !(f.IsGradient || f.IsFunctionGradient) {
+		return fmt.Errorf("function does not implement Grad() nor FuncGrad()")
```

Comment:
"either" -> "or"
"neither" -> "nor"

So in this case "or" (equivalent here to `!(a||b)`).